x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java

```diff
@@ -284,26 +284,30 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
             HttpResponse enrollResponse = null;
             URL enrollNodeUrl = null;
             for (String address : enrollmentToken.getBoundAddress()) {
-                enrollNodeUrl = createURL(new URL("https://" + address), "/_security/enroll/node", "");
-                enrollResponse = client.execute(
-                    "GET",
-                    enrollNodeUrl,
-                    new SecureString(enrollmentToken.getApiKey().toCharArray()),
-                    () -> null,
-                    CommandLineHttpClient::responseBuilder
-                );
-                if (enrollResponse.getHttpStatus() == 200) {
-                    break;
+                try {
+                    enrollNodeUrl = createURL(new URL("https://" + address), "/_security/enroll/node", "");
+                    enrollResponse = client.execute(
+                        "GET",
+                        enrollNodeUrl,
+                        new SecureString(enrollmentToken.getApiKey().toCharArray()),
+                        () -> null,
+                        CommandLineHttpClient::responseBuilder
+                    );
+                    if (enrollResponse.getHttpStatus() == 200) {
+                        break;
+                    }
```

Comment:
The call can fail in two different ways: on the connection or on the HTTP level.

The connection one is to be expected, but the HTTP one is not. Hence I think that a non-200 from the the `node enroll token` endpoint should result in an enrollment failure.

Reply:
>  but the HTTP one is not. Hence I think that a non-200 from the the node enroll token endpoint should result in an enrollment failure.

The enrollment token can contain multiple HTTP addresses for the existing nodes in the cluster. Presumably, each existing node can answer to the enrollment request. Why should a temporary failure in one node ( returning 5xx ) prevent the new node from enrolling, when it can still try and get the enrollment information from another node ?

Reply:
Nevermind me, I forgot we changed this. We call `/_nodes/_local/http` when creating the enrollment token so the IP addresses we get are all for the same node.

Reply:
>  Hence I think that a non-200 from the the node enroll token endpoint should result in an enrollment failure.

It would still result to an enrollment failure _unless_ it was something temporary and there is another bound address for that node so when we retry talking to that address, the node responds successfully. Do you think it's worth exiting on non 200 immediately every time ?

Reply:
The enrollment token only contains the addresses of the node that generated it.

If the endpoint returns 400/500 I'm OK to retry that call, in the hope that it is a temporary failure, but I would say this is not necesary.

Reply:
> The enrollment token only contains the addresses of the node that generated it.

yes, I figured out after writing the comment :) 

> If the endpoint returns 400/500 I'm OK to retry that call, in the hope that it is a temporary failure, but I would say this is not necesary.

No strong opinions, I'll adjust